### PR TITLE
Gnocchi: implement resource types List method

### PR DESCRIPTION
**Pull request by ozerovandrei**
Add new resourcetypes package with List method.
Add tests and a documentation example.

Code references:
https://github.com/gnocchixyz/gnocchi/blob/stable/4.2/gnocchi/rest/api.py#L953
https://github.com/gnocchixyz/gnocchi/blob/stable/4.2/gnocchi/indexer/sqlalchemy.py#L485
https://github.com/gnocchixyz/gnocchi/blob/stable/4.2/gnocchi/resource_type.py#L264

For #54

### CODE REVIEW

File: gnocchi/metric/v1/resourcetypes/results.go [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+
+	// Name is a human-readable resource type identifier.
+	Name string `json:"name"`
+
+	// State represents current status of a resource type.
+	State string `json:"state"`
+}
+
+// ResourceTypeAttribute represents single attribute of a Gnocchi resource type.
+type ResourceTypeAttribute struct {
```
Comment (jtopjian) [inline gnocchi/metric/v1/resourcetypes/results.go]: Can we call this `Attribute`?

**Comment by ozerovandrei** [comment]
@jtopjian Hello.
Attribute was tricky. I haven't started implementing another PR for the #54, because I haven't been sure about current implementation.
Will try to change it according to your request.

**Comment by jtopjian** [comment]
Sounds good. Another option is to leave `Attribute` as a generic `map[string]interface{}`. I guess it depends on if the attributes will be usable or just for informational purposes.

**Comment by ozerovandrei** [comment]
@jtopjian I made changes related to Attribute field. I think that it's better to have at least `Name` and `Type` as separate fields inside the structure because they're common between all attribute types.

**Comment by jtopjian** [comment]
@ozerovandrei This still felt a little strange to me and then I realized it is because we're trying to convert a map of values into an array. 

How about something like this: https://github.com/jtopjian/gophercloud-utils/commit/89b3bbbcc33d84d634a55911e4b44f1f95fd9197

**Comment by ozerovandrei** [comment]
@jtopjian it's because I've tried to do a very strong structure types without any maps from the beginning of that PR. Some Go developer told me to so so once: [#issuecomment-362023884](https://github.com/gophercloud/utils/pull/23#issuecomment-362023884) 😉 

But yeah, I can change this, why not.

**Comment by jtopjian** [comment]
That discussion is still applicable. :)

What I'm proposing is modeling the data in its normal map/object/hash/dict form and not coalescing it into an array/list/slice.

**Comment by jtopjian** [comment]
@ozerovandrei Just to confirm: you're finished with this and it's mergeable?

**Pull request merged**